trustmanager/keyfilestore.go

```diff
@@ -9,60 +9,64 @@ import (
 
 const (
 	keyExtension = "key"
+	aliasExtension = "alias"
 )
 
 // KeyStore is a generic interface for private key storage
 type KeyStore interface {
 	LimitedFileStore
 
-	AddKey(name string, privKey data.PrivateKey) error
+	AddKey(name, alias string, privKey data.PrivateKey) error
 	GetKey(name string) (data.PrivateKey, error)
-	AddEncryptedKey(name string, privKey data.PrivateKey, passphrase string) error
-	GetDecryptedKey(name string, passphrase string) (data.PrivateKey, error)
+	GetKeyAlias(name string) (string, error)
```

Comment:
Conceptually, do we really need to track the "alias" for each key? It seems like we're adding a decent amount of complexity - new files that need to be kept in sync, passed through with imports/exports, etc.

Stepping back, I think the only thing this is used for is to tell the passphraseRetriever what kind of key we're trying to decrypt, so it can check its cache and return the appropriate passphrase. But if we maintained the cache inside notary instead of pushing it out to whatever implements passphrase retrieval, it would be trivial to try all the passphrases we know about, and not have to care what kind of key we want to unlock.

What would you think about replacing the alias stuff with a cache inside the `KeyStore` that keeps track of every passphrase passphraseRetriever has returned and tries those passphrases when it needs to decrypt a key? If one of them works, great. If not, we call passphraseRetriever to prompt the user or check the environment variable. I suspect that would be a lot simpler than storing extra metadata.